editor/plugins/shader_editor_plugin.cpp

```diff
@@ -571,10 +576,20 @@ void ShaderEditorPlugin::_window_changed(bool p_visible) {
 	make_floating->set_visible(!p_visible);
 }
 
+void ShaderEditorPlugin::_file_removed(const String &p_removed_file) {
+	for (uint32_t i = 0; i < edited_shaders.size(); i++) {
+		Ref<Shader> &shader = edited_shaders[i].shader;
+		if (shader->get_meta("_edit_res_path") == p_removed_file) {
```

Comment:
```suggestion
		if (shader->get_path() == p_removed_file) {
```
Resources already know their path.

Reply:
That path is empty because it's already cleared when delete the file in filesystem panel. 
Thus I have to set the meta in `_update_shader_list`, and use the meta to decide which file to close.